blog/release.md

```diff
@@ -0,0 +1,245 @@
+
+### Creating images
+
+When creating a new image, unlike before, image's width and height must be specified.
+
+```ts
+import { Image } from 'image-js';
+
+// Would work before, will throw an error in a new version.
+const image = new Image();
+// Works fine.
+const image2 = new Image(10, 10);
+```
+
+This change makes the Image constructor more explicit by requiring you to specify the dimensions upfront, preventing potential errors from working with uninitialized or undefined-sized images
+
+### Masks
+
+Binary images are now handled by a dedicated `Mask` class instead of Image with `kind: 'BINARY'`.
+
+```ts
+// Before
+const mask = new Image(10, 10, { kind: 'BINARY' });
+```
+
+```ts
+// After
+const mask = new Mask(10, 10);
+```
+
+`Mask` provides better type safety, clearer API, and optimized performance for binary operations.
+
+The new `Mask` class uses 1 byte per pixel (vs 8 pixels per byte), trading ~8x memory usage for significantly faster bit operations and simpler data manipulation.
+
+### Sobel and Scharr filters
+
+[Sobel](https://en.wikipedia.org/wiki/Sobel_operator), [Scharr](https://en.wikipedia.org/wiki/Sobel_operator#Alternative_operators) filters are now combined into a single `derivative()` method.
+
+```ts
+// Before
+const sobelX = img.sobelX();
+const sobelY = img.sobelY();
+
+// After
+const sobelX = img.derivative({ filter: 'sobel' });
+const sobelY = img.derivative({ filter: 'scharr' });
+```
+
+This filter now also accepts only grayscale images, since filters, like Sobel or Scharr, are used mainly on grayscale images to detect edges.
+
+### Method Renaming
+
+Several methods have been renamed for consistency:
+
+**Drawing methods**:
+
+`img.paintPolyline()` ➡️ `img.drawPolyline()`
+
+`img.paintPolygon()` ➡️ `img.drawPolygon()`
+
+`img.paintCircle()` ➡️ `img.drawCircle()`
+
+**Other methods**:
+
+`img.copy()` ➡️ `img.clone()`
+
+`img.clearBit()` ➡️ `img.setBit()`
+
+`img.getLocalMaxima()` ➡️ `img.getExtrema()`
+
+`img.getChannel()` ➡️ `img.extractChannel()`
+
+`img.rotateLeft()` and `img.rotateRight()` ➡️ `img.rotate()`
+
+`img.flipX()` and `img.flipY()` ➡️ `img.flip()`
+
+Consistent naming follows common conventions ("draw\*" for rendering, "clone" for copying objects).
+
+### Compatibility requirements
+
+- Node.js: 18+ (previously 14+)
+- TypeScript: 5.2.2+ (if using TypeScript)
+
+### Removed Features
+
+The following deprecated features have been removed:
+
+- `countAlphaPixel()` - Use custom pixel counting with `getPixel()`
+- `paintLabels()` - Feature was removed due to dependency issues. We plan to add it back in the future updates.
+- `warpingFourPoints()` - Use `getPerspectiveWarp()` + `transform()`.
+- 32-bit color depth has been currently deprecated. We plan to add it back in the future updates as well.
+- `CMYK` and `HSL` color models have been deprecated. We plan to add it in the future updates.
+- `insert()` has been deprecated. We plan to add it in the future updates.
```

Comment:
These two points have been discussed before. But I can't remember the reason why they were not added.